src/main/java/ch/njol/skript/sections/EffSecShoot.java

```diff
@@ -0,0 +1,215 @@
+	"You can modify the entity in this section, using for example 'event-entity' or 'event-arrow'.",
+	"Do note that other event values, such as 'player', won't work in this section."})
+@Examples({"shoot an arrow",
+	"make the player shoot a creeper at speed 10",
+	"shoot a pig from the creeper",
+	"",
+	"# Shoot section example",
+	"command /firearrow:",
+	"\ttrigger:",
+	"\t\tshoot an arrow:",
+	"\t\t\tignite event-projectile"})
+@Since("1.4, INSERT VERSION (with section)")
+public class EffSecShoot extends EffectSection {
+
+	public static class ShootEvent extends Event {
+
+		private final Entity entity;
+
+		public ShootEvent(Entity entity) {
+			this.entity = entity;
+		}
+
+		public Entity getEntity() {
+			return entity;
+		}
+
+		@Override
+		@NotNull
+		public HandlerList getHandlers() {
+			throw new IllegalStateException();
+		}
+
+	}
+
+	static {
+		Skript.registerSection(EffSecShoot.class,
+			"shoot %entitydatas% [from %livingentities/locations%] [(at|with) (speed|velocity) %-number%] [%-direction%]",
+			"(make|let) %livingentities/locations% shoot %entitydatas% [(at|with) (speed|velocity) %-number%] [%-direction%]");
+		EventValues.registerEventValue(ShootEvent.class, Entity.class, new Getter<Entity, ShootEvent>() {
+			@Override
+			public Entity get(ShootEvent shootEvent) {
+				return shootEvent.getEntity();
+			}
+		}, 0);
+	}
+
+	private static final Double DEFAULT_SPEED = 5.0;
+
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private Expression<EntityData<?>> types;
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private Expression<?> shooters;
+	@Nullable
+	private Expression<Number> velocity;
+	@Nullable
+	private Expression<Direction> direction;
+
+	@Nullable
+	public static Entity lastSpawned = null;
+
+	@Nullable
+	private Trigger trigger;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs,
+						int matchedPattern,
+						Kleenean isDelayed,
+						ParseResult parseResult,
+						@Nullable SectionNode sectionNode,
+						@Nullable List<TriggerItem> triggerItems) {
+		types = (Expression<EntityData<?>>) exprs[matchedPattern];
+		shooters = exprs[1 - matchedPattern];
+		velocity = (Expression<Number>) exprs[2];
+		direction = (Expression<Direction>) exprs[3];
+
+		if (sectionNode != null) {
+			trigger = loadCode(sectionNode, "shoot", ShootEvent.class);
+		}
+
+		return true;
+	}
+
+	@Override
+	@Nullable
+	@SuppressWarnings({"unchecked", "rawtypes"})
+	protected TriggerItem walk(Event e) {
+		lastSpawned = null;
+
+		Object localVars = Variables.copyLocalVariables(e);
+
+		Consumer<? extends Entity> consumer;
+		if (trigger != null) {
+			consumer = o -> {
+				lastSpawned = o;
+				ShootEvent shootEvent = new ShootEvent(o);
+				// Copy the local variables from the calling code to this section
+				Variables.setLocalVariables(shootEvent, localVars);
+				trigger.execute(shootEvent);
```

Comment:
```suggestion
				trigger.execute(shootEvent);
				Variables.setLocalVariables(e, localVars); // Carry over variable changes to the rest of the main trigger
```
This is something that needs addressed in these "spawn" type sections.
Basically, variable changes made within a Section like this should be reflected for the rest of the trigger outside of the section.

I would make sure to add some testing for this in your test script.